### PR TITLE
Misc fixes around API permissions, global settings and template UX

### PR DESCRIPTION
**Pull request by rhtyd**
- This allows getUploadParamsForIso for all user role types, also fixes
authorised field for getUploadParamsForTemplate API.
- Fix global setting description to say what is used when value is empty/blank.
- For VM running/allocated usage description, use parenthesis to return the instance name and ID.
- Display template download progress when template is added to a project

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
Pinging for review @shwstppr @anuragaw

File: api/src/main/java/org/apache/cloudstack/query/QueryService.java [hunk 1]
```diff
@@ -92,7 +92,7 @@
 
     static final ConfigKey<String> UserVMBlacklistedDetails = new ConfigKey<String>("Advanced", String.class,
             "user.vm.blacklisted.details", "rootdisksize, cpuOvercommitRatio, memoryOvercommitRatio, Message.ReservedCapacityFreed.Flag",
-            "Determines whether users can view certain VM settings", true);
+            "Determines whether users can view certain VM settings. When set to empty, default value used is: rootdisksize, cpuOvercommitRatio, memoryOvercommitRatio, Message.ReservedCapacityFreed.Flag.", true);
```
Comment (anuragaw) [inline api/src/main/java/org/apache/cloudstack/query/QueryService.java]: Slight addition could be there optinally - "To allow all settings to be visible set this to `none`"?

Comment (rhtyd) [reply api/src/main/java/org/apache/cloudstack/query/QueryService.java]: The main issue with setting to a fake value is that, what is the none is used as a valid detail (most likely that's never going to happen but for sanity-sake). Which is why I chose to simply state that behaviour.

Comment (anuragaw) [reply api/src/main/java/org/apache/cloudstack/query/QueryService.java]: Sounds good.

**Comment by borisstoyanov** [comment]
These changes seem fine, there's one problem related to uploading template and iso from local, which I'm not sure if it will be fixed in this PR? 
cc: @rhtyd

**Comment by rhtyd** [comment]
I'll check/test @borisstoyanov

**Comment by anuragaw** [comment]
@borisstoyanov, @rhtyd - I also see a format not matching the selected format when I upload a local macchina qcow2 template on KVM based monkey box. Investigating bit more.

**Comment by rhtyd** [comment]
For the upload iso/template fix, I'll send another PR Bobby. Thanks.
Merging based on test results and lgtms.

**Pull request merged**